### DISCUSSION TITLE
GitOps: authentication in private Gitlab repo

### DISCUSSION THREAD

**Discussion by mr-rootmann** [Apicurio/apicurio-registry#4962] [Q&A]
Greetings,

I am currently struggling with figuring out how the GitOps setup of ApiCurio Registry works.

After some hours, I managed to find all supported environment variables for GitOps setup:
APICURIO_STORAGE_KIND=gitops
APICURIO_GITOPS_ID=main
APICURIO_GITOPS_REPO_ORIGIN_URI=https://gitlab.com/*
APICURIO_GITOPS_REPO_ORIGIN_BRANCH=develop
APICURIO_GITOPS_REFRESH_EVERY=30s
APICURIO_GITOPS_WORKDIR=*

Please add these to README file. This could save someone's time. 
In general the GitOps setup is barely documented. 

And there is no authentication env var. Why? 

As a result I get this error when trying to run a Docker container:

Caused by: org.eclipse.jgit.api.errors.TransportException: https://gitlab.com/*: Authentication is required but no CredentialsProvider has been registered
        at org.eclipse.jgit.api.FetchCommand.call(FetchCommand.java:249)
        at io.apicurio.registry.storage.impl.gitops.GitManager.poll(GitManager.java:144)
        at io.apicurio.registry.storage.impl.gitops.GitManager_ClientProxy.poll(Unknown Source)
        at io.apicurio.registry.storage.impl.gitops.GitOpsRegistryStorage.refresh(GitOpsRegistryStorage.java:132)

Was the topic of authentication in Gitlab/*hub repo considered at all? 
I did not find any hint in the source code for that. 

FYI, I am running apicurio/apicurio-registry:latest (3.0.0.M3)

**Comment by EricWittmann** [comment]
The GitOps work is all very experimental.  We definitely haven't created documentation for it yet, or examples, or anything like that.  Even when we release 3.0.0, the GitOps feature will remain experimental.  It's something we think is very valuable, but other things have had a higher priority.  So we continue to explore gitops when we have time.

@jsenko can probably provide more context here.

**Comment by jsenko** [comment]
Hello, as Eric mentioned, the GitOps setup is barely documented because it's an experimental feature and is not ready for production. Specifically, the authentication is not implemented or tested, that's why you're getting the error. While we want to fully support GitOps in medium-future, other features have priority at the moment. However if you're interested in contributing the README or authentication support, we'll be happy to review.

**Reply by EricWittmann** [reply]
Yes we'd be happy to have any contributions in this area - and are also very happy to help with guidance (e.g. pointing to where in the code some feature would need to be added).

**Reply by that-is-not-a-bug** [reply]
Can you please share what is missing and how you were planning to implement this feature?
